logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt

```diff
@@ -127,14 +145,20 @@ internal class CallDataSource(
         )
     }
 
-    override fun updateCallStatusById(conversationId: String, status: CallStatus) {
+    override suspend fun updateCallStatusById(conversationId: String, status: CallStatus) {
         val callProfile = _callProfile.value
         val modifiedConversationId = conversationId.toConversationId().toString()
         callProfile.calls[modifiedConversationId]?.let { call ->
             val updatedCalls = callProfile.calls.toMutableMap().apply {
+                val establishedTime =
+                    if (status == CallStatus.ESTABLISHED) timeParser.currentTimeStamp()
+                    else call.establishedTime
+
                 this[modifiedConversationId] = call.copy(
-                    status = status
+                    status = status,
+                    establishedTime = establishedTime
                 )
+                persistMissedCallMessageIfNeeded(this[modifiedConversationId]!!)
```

Comment:
We need also to cover with unit test the case when `persistMessage` is not invoked